specs/remove-hypervisor-dependency.md

```diff
@@ -0,0 +1,45 @@
+# Remove hypervisor dependencies
+
+## Scope
+We want to make Open vStorage a real storage product which no longer depends on a hypervisor. In previous versions we always assumed that the Volume Driver ran together with a hypervisor (VMware/KVM). With the Edge this no longer the case as the edge client runs on the compute nodes (hypervisor) and the volume driver runs separately on a different node.
+With th FR we want to break this hypervisor dependency. This means you can now install Open vStorage and create a vdisk and add the hypervisor later.
+
+## OVS setup
+* Remove the question(s) and code about the hypervisor (VMware and ESXi)
+
+## Hypervisors
+### NFS
+* Remove the NFS path for VMware (VMware will be supported through iSCSI)
+### KVM
+* For KVM we need to still support the FUSE filesystem approach
+* When creating a vDisk on the fuse layer we don't interact with th hypervisor to find out which VM it is attached too.
+* .raw only
+### Edge
+* The Edge should be able to work without the FUSe layer
```

Comment:
Correct. We want to remove the fuse in a later stage
Op 25 jun. 2016 10:04 a.m. schreef "Kenneth Henderick" <
notifications@github.com>:

> In specs/remove-hypervisor-dependency.md
> <https://github.com/openvstorage/framework/pull/668#discussion_r68485938>:
>
> > +## Scope
> > +We want to make Open vStorage a real storage product which no longer depends on a hypervisor. In previous versions we always assumed that the Volume Driver ran together with a hypervisor (VMware/KVM). With the Edge this no longer the case as the edge client runs on the compute nodes (hypervisor) and the volume driver runs separately on a different node.
> > +With th FR we want to break this hypervisor dependency. This means you can now install Open vStorage and create a vdisk and add the hypervisor later.
> > +
> > +## OVS setup
> > +* Remove the question(s) and code about the hypervisor (VMware and ESXi)
> > +
> > +## Hypervisors
> > +### NFS
> > +* Remove the NFS path for VMware (VMware will be supported through iSCSI)
> > +### KVM
> > +* For KVM we need to still support the FUSE filesystem approach
> > +* When creating a vDisk on the fuse layer we don't interact with th hypervisor to find out which VM it is attached too.
> > +* .raw only
> > +### Edge
> > +* The Edge should be able to work without the FUSe layer
>
> I think @wimpers <https://github.com/wimpers> meant that the framework
> should try to not rely on the FUSE mountpoint as much as possible.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openvstorage/framework/pull/668/files/716b53a8dd494dba50541f7a58977dd6d12be4ef#r68485938>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIAQ35MHsCmZ5PzDKCyozYYoIQU3lFGaks5qPOD1gaJpZM4I9org>
> .
>